addons/account/models/account_account.py

```diff
@@ -1040,194 +1041,281 @@ def get_import_templates(self):
+            else:
+                table = field_to_update.relation_table
+                account_column = field_to_update.column2
+                model_column = field_to_update.column1
+            records_to_update = self.env[model].search([(field_to_update.name, '=', self.id)])
+            records_to_update.invalidate_recordset([field_to_update.name])
+
+            # We use a subquery + _field_to_sql in order to handle related company_id fields.
+            company_id_query = self.env[model]._where_calc([(field_to_update.name, '=', self.id)])
+            company_id_select = company_id_query.select(
+                SQL('%s AS id', self.env[model]._field_to_sql(company_id_query.table, 'id')),
+                SQL('%s AS company_id', self.env[model]._field_to_sql(company_id_query.table, company_id_field, company_id_query)),
+            )
+            query = SQL(
+                """
+                 UPDATE %(table)s
+                    SET %(account_column)s = (
+                            %(new_account_id_by_company_id_json)s::jsonb->>
+                            table_with_company_id.%(company_id_column)s::text
+                        )::int
+                   FROM (%(company_id_select)s) table_with_company_id
+                  WHERE table_with_company_id.id = %(model_column)s
+                    AND %(table)s.%(account_column)s = %(account_id)s
+                    AND table_with_company_id.%(company_id_column)s IN %(company_ids_to_update)s
+                """,
+                table=SQL.identifier(table),
+                account_column=SQL.identifier(account_column),
+                new_account_id_by_company_id_json=new_account_id_by_company_id_json,
+                company_id_column=SQL.identifier(company_id_field),
+                company_id_select=company_id_select,
+                model_column=SQL.identifier(table, model_column),
+                account_id=self.id,
+                company_ids_to_update=tuple(new_account_id_by_company_id),
+            )
+            self.env.cr.execute(query)
+
+        # 3.2: Add Reference fields that reference account.account
+        # Include the 'value_reference' field on ir.property that is nominally a char (used e.g. for default receivable accounts)
+        reference_fields = self.env['ir.model.fields'].search([
+            '|',
+            '&', ('ttype', '=', 'reference'), ('store', '=', True),
+            '&', ('model', '=', 'ir.property'), ('name', '=', 'value_reference'),
+        ])
+        for field_to_update in reference_fields:
+            model = field_to_update.model
+            if not self.env[model]._auto:
+                continue
+            if model == 'res.company':
+                company_id_field = 'id'
+            elif 'company_id' in self.env[model]:
+                company_id_field = 'company_id'
+            else:
+                continue
+            if not self.env[model]._fields[company_id_field].store:
+                continue
```

Comment:
Same code as above, maybe extract in a function that takes the model as parameter and returns the field.
If it doesn't return a field, then `continue`
```suggestion
            if not (company_id_field := get_company_field(model)):
                continue
```